Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -354,6 +354,8 @@ class Connection : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<Connec
     void removeMessageReceiver(ReceiverName, uint64_t destinationID = 0);
 
     bool open(Client&, SerialFunctionDispatcher& = RunLoop::current());
+    // Ensures that messages sent prior to the call are not affected by invalidate() or crash done after the call returns.
+    Error flushSentMessages(Timeout);
```

Comment:
I would prefer this to be an enum flag to `invalidate` (plus the optional timeout param I guess).

That would make it easier to know what the behaviour is for callsites that aren't using the flush behaviour, and forces new code to make an explicit decision (rather than potentially just forgetting to add a flush call).

Reply:
Well, essentially no callsite ever should invalidate without flush.
Invalidate cannot flush, because invalidate should not fail but flush can time out.
I'd like to add the flushes individually, in separate commits to prevent any reverts/bureaucracy.
when all invalidate callsites have the flush, I'd then consider lumping them.